### DISCUSSION TITLE
History of msgs

### DISCUSSION THREAD

**Discussion by take2make** [facebookresearch/ParlAI#3696] [Q&A] [answered]
How can I add history to a new bot? For example, yesterday I was talking with a bot but today he does not recognize me. To eliminate the problem I saved all message history, but how to add it to a bot?

**Comment by stephenroller** [comment, marked as answer]
the easiest way is to concatenate it all and have the agent `observe()` it

**Comment by renzs6** [comment]
Hi @stephenroller what's the expected format of the concatenated dialogue history? is it just a plain list ["bot: hello", "user:"hi", "bot": "how are you?",etc.]

**Reply by klshuster** [reply]
For concatenating you'd want to turn the history into one full string, with a delimiter separating the utterances.
